### DISCUSSION TITLE
YouTube video playbacks are stalling.

### DISCUSSION THREAD

**Discussion by nobara256** [Joshua-Riek/ubuntu-rockchip#159] [General]
If you leave a video playing and switch to a different tab or minimise chromium window, after a while the video stalls while the audio plays normally when the YouTube tab or window is focused back. Anything at or above 1080p can certainly trigger this most of the times. Sometimes a single CPU core hits 100% during this stall. I have faced this issue in previous builds too.

Here is the application log after such stall: [log.txt](https://github.com/Joshua-Riek/ubuntu-rockchip/files/11532333/log.txt)

Also skipping to random portions of a 1080p video can take forever to load and sometimes I need to refresh the tab.

**Comment by Joshua-Riek** [comment]
That is likely due to rkmpp continuously crashing. Something to do with the chromium hack from Rockchip to support accelerated video playback. It has a number of issues but nothing I can do about it.

**Reply by nobara256** [reply]
Thanks for explaining the issue. At least now I know it's a chromium issue.

**Comment by nobara256** [comment]
Interesting. Just received a bunch of package updates today, among them were `librockchip-mpp-dev` & `librockchip-mpp1` and some `mali` packages. I tried to run few 1080p@60 fps videos on YouTube and now the videos doesn't appear to stall frequently like before even if they stall they recover after a few seconds. Also no more hanging while skipping to random portions of the video.
Overall, YouTube appears to be more snappy.


meanwhile dmesg is flooded with:
`mpp_rkvdec2 fdc38100.rkvdec-core: resetting...`
`mpp_rkvdec2 fdc38100.rkvdec-core: reset done`
